src/modules/shortcut_guide/overlay_window.cpp

```diff
@@ -503,15 +505,33 @@ void D2DOverlayWindow::hide_thumbnail() {
 
 void D2DOverlayWindow::render(ID2D1DeviceContext5* d2d_dc) {
   if (!winkey_held() || is_start_visible()) {
-    hide();
-    instance->was_hidden();
-    return;
+	  auto current_anim_value = animation.value(Animation::AnimFunctions::LINEAR);
+	  if (!hiding) { // when user is done viewing the overlay
+		  animation.reset(anim_time * current_anim_value);
+		  hiding = true;
+	  }
+	  else if (current_anim_value == 1.0) { // animation to hide overlay has finished
+		  hide();
+		  instance->was_hidden();
+		  animation.reset(anim_time);
+		  hiding = false;
+		  return;
+	  }
   }
   d2d_dc->Clear();
   int x_offset = 0, y_offset = 0, dimention = 0;
+  int alpha;
+  double pos_anim_value;
   auto current_anim_value = (float)animation.value(Animation::AnimFunctions::LINEAR);
-  SetLayeredWindowAttributes(hwnd, 0, (int)(255*current_anim_value), LWA_ALPHA);
-  double pos_anim_value = 1 - animation.value(Animation::AnimFunctions::EASE_OUT_EXPO);
+  if (hiding) {
+	  alpha = 255 * (1.0 - current_anim_value);
```

Comment:
I get ` warning C4244: '=': conversion from 'double' to 'int', possible loss of data` on this line.